pkgs/applications/window-managers/jwm/default.nix

```diff
@@ -1,22 +1,33 @@
-{ stdenv, fetchurl, libX11, libXext, libXinerama, libXpm, libXft, freetype,
+{ stdenv, fetchurl, pkgconfig, automake, autoconf, libtool, libX11, libXext,
+  libXinerama, libXpm, libXft, which, xorg, libXau, libXdmcp, libpng, libjpeg,
+  expat, xproto, xextproto, xineramaproto, librsvg, gettext, freetype,
   fontconfig }:
 
 stdenv.mkDerivation rec {
-  name = "jwm-2.2.2";
+  name = "jwm-${version}";
+  version = "s1406";
   
   src = fetchurl {
-     url = "http://www.joewing.net/projects/jwm/releases/${name}.tar.xz";
-     sha256 = "0nhyy78c6imk85d47bakk460x0cfhkyghqq82zghmb00dhwiryln";
+     url = "https://github.com/joewing/jwm/archive/${version}.tar.gz";
+     sha256 = "0yk22b7cshhyfpcqnb4p59yxspx95xg9yp1kmkxi2fyw95cacab4";
   };
 
-  buildInputs = [ libX11 libXext libXinerama libXpm libXft freetype 
-    fontconfig ];
+  buildInputs = [
+    pkgconfig automake autoconf libtool libX11 libXext libXinerama libXpm
+    libXft which xorg.libXrender libXau libXdmcp libpng libjpeg expat xproto
+    xextproto xineramaproto librsvg gettext freetype fontconfig
+  ];
 
   preConfigure = ''
-    export NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -I${freetype}/include/freetype2 "
-    export NIX_LDFLAGS="$NIX_LDFLAGS -lXft -lfreetype -lfontconfig "
+    ./autogen.sh
   '';
 
+  configureFlags = [
+    "--enable-shape"
+    "--enable-xrender"
+    "--enable-png"
+  ];
+
   postInstall =
     ''
       sed -i -e s/rxvt/xterm/g $out/etc/system.jwmrc
```

Comment:
This derivation seems pretty old you might want to check if these patches are still needed.